HLTrigger/Configuration/scripts/hltListPaths

```diff
@@ -1,41 +1,84 @@
-    return '--%s --%s --configName %s' % (menu.database, menu.version, menu.name)
+    configline = f'--{config.menu.database} --{config.menu.version} --configName {config.menu.name}'
 
-def getPathList(config):
-  cmdline = 'hltConfigFromDB --cff %s --noedsources --noes --noservices --nosequences --nomodules' % _build_query(config.menu)
+  # cmd to download HLT configuration
+  cmdline = f'hltConfigFromDB {configline} --noedsources --noes --noservices'
   if config.proxy:
-     cmdline += f" --dbproxy --dbproxyhost {config.proxy_host} --dbproxyport {config.proxy_port}"
-  data = pipe.pipe(cmdline).decode()
-  if 'Exhausted Resultset' in data or 'CONFIG_NOT_FOUND' in data:
-    raise ImportError('%s is not a valid HLT menu' % menu.value)
+    cmdline += f' --dbproxy --dbproxyhost {config.proxy_host} --dbproxyport {config.proxy_port}'
 
+  # load HLT configuration
+  try:
+    foo = {'process': None}
+    exec(pipe.pipe(cmdline).decode(), foo)
+    process = foo['process']
+  except:
+    raise Exception(f'query did not return a valid python file:\n query="{cmdline}"')
+
+  if not isinstance(process, cms.Process):
+    raise Exception(f'query did not return a valid HLT menu:\n query="{cmdline}"')
+
+  # Paths only
   if config.selection == 'paths':
-    filter = re.compile(r' *= *cms.Path.*')
+    pathDict = process.paths_()
+
+  # EndPaths only
   elif config.selection == 'endpaths':
-    filter = re.compile(r' *= *cms.EndPath.*')
-  else: # 'all'
-    filter = re.compile(r' *= *cms.(End)?Path.*')
+    pathDict = process.endpaths_()
 
-  paths  = [ filter.sub('', line) for line in data.splitlines() if filter.search(line) ]
-  return paths
+  # FinalPaths only
+  elif config.selection == 'finalpaths':
+    pathDict = process.finalpaths_()
 
+  # Paths, EndPaths, and FinalPaths ('all')
+  else:
+    pathDict = zip(process.paths_(), process.endpaths_(), process.finalpaths_())
+
+  ret = []
+  for pathName in pathDict:
+
+    # skip if name of the path matches any of
+    # the regular expressions listed in "--exclude"
+    skipPath = False
+    for excludeRegExpr in config.excludeRegExprs:
+      if bool(re.search(excludeRegExpr, pathName)):
+        skipPath = True
+        break
+    if skipPath:
+      continue
+
+    if config.no_dependent_paths:
+      # do not include "dependent paths", i.e. paths that depend on the result of other paths in the same job
+      # the current criterion to identify a path as "dependent" is that
+      # (1) the path contains a "TriggerResultsFilter" module and
+      # (2) the latter module uses the TriggerResults of the current process, and has a non-empty list of "triggerConditions"
+      path = pathDict[pathName]
+      pathIsDependent = False
+      for moduleName in path.moduleNames():
+        module = getattr(process, moduleName)
+        if module.type_() == 'TriggerResultsFilter' \
+          and hasattr(module, 'hltResults') and (module.hltResults.getProcessName() in ['', '@currentProcess']) \
```

Comment:
Why ```''``` in here? That would refer to the triggerresults of a previous process, so would formally be ok...
What about checking parameter ```usePathStatus == true```?

Reply:
I believe cms.EndPaths are/should not be considered at all, so we are left with normal paths (ie, incl. Dataset paths)

Reply:
cms.FinalPaths should also not be considered in integration tests.

Reply:
as we are after comparing physics trigger paths only

Reply:
Sure, indeed we use `hltListPaths -p` (Paths only) inside the integration tests. At the same time, the script `hltListPaths` in itself supports considering EndPaths and FinalPaths, so I was trying to maintain the script general enough.